python/cudf/cudf/core/dataframe.py

```diff
@@ -2049,27 +2049,22 @@ def _make_operands_and_index_for_binop(
             if (
                 not can_reindex
                 and fn in cudf.utils.utils._EQUALITY_OPS
-                and (
-                    not self._data.to_pandas_index().equals(
-                        other.index.to_pandas()
-                    )
-                )
+                and not (
```

Comment:
I think we need to revert my suggestion, because from the diff I thought everything below was inside this `if` block, but they are outside of it.

Reply:
I just had to move the walrus condition to the top so the `if` branch doesn't short circuit without defining these variables

Reply:
Moving the condition looked to have done the trick and all green here